cabal-testsuite/PackageTests/ConditionalAndImport/cabal.test.hs

```diff
@@ -1,6 +1,10 @@
 import Test.Cabal.Prelude
 main = cabalTest $
   withRepo "repo" $ do
+     -- this shouldn't succeed, but it does, because this doesn't override path correctly, sigh.
+     fails $ withEnv [("PATH",Just "")] $ cabal "v2-sdist" []
```

Comment:
What about trying a garbage PATH like `Just "foo"`?  What happens then?

Reply:
doesn't seem to help. cabal still finds ghc. Seems like "withCreateProcess" isn't properly overriding the PATH, so this is not just a testsuite issue, sigh.

Reply:
I've bumped process one more notch a couple of days ago. Perhaps it would help to rebase and retry?

Reply:
or rather, if i end up invoking bash -c "echo $PATH" then it does print the modified path. but somehow there's a cache or something so it doesn't fail here, not sure, but terribly frustrating.